kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/async/AsyncProcessor.java

```
@@ -27,15 +28,22 @@
 @Slf4j
 @SuppressWarnings("UnstableApiUsage")
 public abstract class AsyncProcessor<K, V, KOUT, VOUT> implements Processor<K, V, KOUT, VOUT> {
-
-  private final Executor executor;
+  private List<Executor> executorList;
   private final BlockingQueue<CompletableFuture<List<RecordToForward<KOUT, VOUT>>>> pendingFutures;
   private final RateLimiter rateLimiter;
+  private final Optional<KeyToAsyncThreadPartitioner<K>> mayBeKeyToAsyncThreadPartitioner;
   private ProcessorContext<KOUT, VOUT> context;
 
   public AsyncProcessor(
-      Supplier<Executor> executorSupplier, AsyncProcessorConfig asyncProcessorConfig) {
-    this.executor = executorSupplier.get();
+      Supplier<List<Executor>> executorListSupplier,
+      AsyncProcessorConfig asyncProcessorConfig,
+      Optional<KeyToAsyncThreadPartitioner<K>> mayBeKeyToAsyncThreadPartitioner) {
+    this.executorList = executorListSupplier.get();
+    this.mayBeKeyToAsyncThreadPartitioner = mayBeKeyToAsyncThreadPartitioner;
+    if (executorList.size() == 1 && mayBeKeyToAsyncThreadPartitioner.isPresent()) {
+      log.warn(
```

should we throw instead of warn log to not have hidden misses?